The job **`aarch64-gnu-llvm-20-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/21174471698/job/60899468453) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/60899468453) [(plain)](https://github.com/rust-lang/rust/commit/0f242ef478d237638b1dac4a01666d3c450641da/checks/60899468453/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [ui] tests/ui/imports/ambiguous-2.rs ... ok
test [ui] tests/ui/imports/ambiguous-4.rs ... ok
test [ui] tests/ui/imports/ambiguous-7.rs ... ok
test [ui] tests/ui/imports/ambiguous-9.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-glob-vs-multiouter.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-globvsglob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-no-implicit-prelude.rs ... ok
test [ui] tests/ui/imports/ambiguous-8.rs ... ok
test [ui] tests/ui/imports/ambiguous-glob-vs-expanded-extern.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-core-glob.rs ... ok
test [ui] tests/ui/imports/ambiguous-panic-non-prelude-std-glob.rs ... ok
---
To only update this specific test, also pass `--test-args suggestions/derive-trait-for-method-call.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/suggestions/derive-trait-for-method-call.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/suggestions/derive-trait-for-method-call" "-A" "unused" "-W" "unused_attributes" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0599]: the method `test` exists for struct `Foo<Enum, CloneEnum>`, but its trait bounds were not satisfied
##[error]  --> /checkout/tests/ui/suggestions/derive-trait-for-method-call.rs:28:15
   |
LL | enum Enum {
   | --------- doesn't satisfy `Enum: Clone` or `Enum: Default`
...
LL | enum CloneEnum {
   | -------------- doesn't satisfy `CloneEnum: Default`
...
LL | struct Foo<X, Y> (X, Y);
   | ---------------- method `test` not found for this struct
...
LL |     let y = x.test();
   |               ^^^^ method cannot be called on `Foo<Enum, CloneEnum>` due to unsatisfied trait bounds
   |
note: the following trait bounds were not satisfied:
      `CloneEnum: Default`
      `Enum: Clone`
      `Enum: Default`
  --> /checkout/tests/ui/suggestions/derive-trait-for-method-call.rs:20:9
   |
LL | impl<X: Clone + Default + , Y: Clone + Default> Foo<X, Y> {
   |         ^^^^^   ^^^^^^^                ^^^^^^^  ---------
   |         |       |                      |
   |         |       |                      unsatisfied trait bound introduced here
   |         |       unsatisfied trait bound introduced here
   |         unsatisfied trait bound introduced here
help: consider annotating `CloneEnum` with `#[derive(Default)]`
---
...
LL | struct CloneStruct {
   | ------------------ doesn't satisfy `CloneStruct: Default`
...
LL | struct Foo<X, Y> (X, Y);
   | ---------------- method `test` not found for this struct
...
LL |     let y = x.test();
   |               ^^^^ method cannot be called on `Foo<Struct, CloneStruct>` due to unsatisfied trait bounds
   |
note: the following trait bounds were not satisfied:
      `CloneStruct: Default`
      `Struct: Clone`
      `Struct: Default`
  --> /checkout/tests/ui/suggestions/derive-trait-for-method-call.rs:20:9
   |
LL | impl<X: Clone + Default + , Y: Clone + Default> Foo<X, Y> {
   |         ^^^^^   ^^^^^^^                ^^^^^^^  ---------
   |         |       |                      |
   |         |       |                      unsatisfied trait bound introduced here
   |         |       unsatisfied trait bound introduced here
   |         unsatisfied trait bound introduced here
help: consider annotating `CloneStruct` with `#[derive(Default)]`
---
LL + #[derive(Clone, Default)]
LL | struct Struct {
   |

error[E0599]: the method `test` exists for struct `Foo<Vec<Enum>, Instant>`, but its trait bounds were not satisfied
##[error]  --> /checkout/tests/ui/suggestions/derive-trait-for-method-call.rs:40:15
   |
LL | enum Enum {
   | --------- doesn't satisfy `Enum: Clone`
...
LL | struct Foo<X, Y> (X, Y);
   | ---------------- method `test` not found for this struct
...
LL |     let y = x.test();
   |               ^^^^ method cannot be called on `Foo<Vec<Enum>, Instant>` due to unsatisfied trait bounds
   |
note: trait bound `Instant: Default` was not satisfied
  --> /checkout/tests/ui/suggestions/derive-trait-for-method-call.rs:20:40
   |
LL | impl<X: Clone + Default + , Y: Clone + Default> Foo<X, Y> {
   |                                        ^^^^^^^  ---------
   |                                        |
   |                                        unsatisfied trait bound introduced here
   = note: the following trait bounds were not satisfied:
           `Enum: Clone`
           which is required by `Vec<Enum>: Clone`
help: consider annotating `Enum` with `#[derive(Clone)]`
   |
LL + #[derive(Clone)]
LL | enum Enum {
   |

```

</details>

--- a/compiler/rustc_hir_typeck/src/method/suggest.rs
+++ b/compiler/rustc_hir_typeck/src/method/suggest.rs
@@ -3300,8 +3300,8 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
             };
             if let Some(diagnostic_name) = self.tcx.get_diagnostic_name(trait_pred.def_id()) {
                 let can_derive = match diagnostic_name {
-                    sym::Default => !adt.is_enum(),
-                    sym::Eq
+                    sym::Default
+                    | sym::Eq
                     | sym::PartialEq
                     | sym::Ord
                     | sym::PartialOrd

--- a/tests/ui/suggestions/derive-trait-for-method-call.stderr
+++ b/tests/ui/suggestions/derive-trait-for-method-call.stderr
@@ -25,11 +25,14 @@ LL | impl<X: Clone + Default + , Y: Clone + Default> Foo<X, Y> {
    |         |       |                      unsatisfied trait bound introduced here
    |         |       unsatisfied trait bound introduced here
    |         unsatisfied trait bound introduced here
-note: the trait `Default` must be implemented
-  --> $SRC_DIR/core/src/default.rs:LL:COL
-help: consider annotating `Enum` with `#[derive(Clone)]`
+help: consider annotating `CloneEnum` with `#[derive(Default)]`
    |
-LL + #[derive(Clone)]
+LL + #[derive(Default)]
+LL | enum CloneEnum {
+   |
+help: consider annotating `Enum` with `#[derive(Clone, Default)]`
+   |
+LL + #[derive(Clone, Default)]
 LL | enum Enum {
    |
 

--- a/tests/ui/suggestions/suggest-derive-default-for-enums.rs
+++ b/tests/ui/suggestions/suggest-derive-default-for-enums.rs
@@ -1,0 +1,15 @@
+// A test showing that we suggest deriving `Default` for enums.
+enum MyEnum {
+    A,
+}
+
+trait Foo {
+    fn bar(&self) {}
+}
+impl<T: Default> Foo for T {}
+
+fn main() {
+    let x = MyEnum::A;
+    x.bar();
+    //~^ ERROR the method `bar` exists for enum `MyEnum`, but its trait bounds were not satisfied
+}

--- a/tests/ui/suggestions/suggest-derive-default-for-enums.stderr
+++ b/tests/ui/suggestions/suggest-derive-default-for-enums.stderr
@@ -1,0 +1,25 @@
+error[E0599]: the method `bar` exists for enum `MyEnum`, but its trait bounds were not satisfied
+  --> $DIR/suggest-derive-default-for-enums.rs:13:7
+   |
+LL | enum MyEnum {
+   | ----------- method `bar` not found for this enum because it doesn't satisfy `MyEnum: Default` or `MyEnum: Foo`
+...
+LL |     x.bar();
+   |       ^^^ method cannot be called on `MyEnum` due to unsatisfied trait bounds
+   |
+note: trait bound `MyEnum: Default` was not satisfied
+  --> $DIR/suggest-derive-default-for-enums.rs:9:9
+   |
+LL | impl<T: Default> Foo for T {}
+   |         ^^^^^^^  ---     -
+   |         |
+   |         unsatisfied trait bound introduced here
+help: consider annotating `MyEnum` with `#[derive(Default)]`
+   |
+LL + #[derive(Default)]
+LL | enum MyEnum {
+   |
+
+error: aborting due to 1 previous error
+
+For more information about this error, try `rustc --explain E0599`.

